### PR TITLE
[LOGONUI] Make Logon UI compilable again

### PR DESCRIPTION
**Pull request by SigmaTel71**
## Purpose

Logon UI was excluded from compilation queue in CMake, probably for not being compliant with 'warnings are errors' standard active since some moment. In future, I would probably try to make Logon UI usable alongside with classic msgina.

JIRA issue: None

## Proposed changes

- include /system/logonui/ subdirectory
- change `wcscpy_s` to `StringCchCopyW` because it's implementation is not compatible yet with ReactOS
- exclude GetULONG() and GetULONG2() as they are unused

### CODE REVIEW

File: base/system/logonui/NT5design.c [hunk 1]
```diff
@@ -136,8 +136,9 @@ NT5_DrawBaseBackground(HDC hdcDesktop)
 {
     HBITMAP hBitmap = NULL;
     HDC hdcMem = NULL;
+    #if 0
```
Comment (binarymaster) [inline base/system/logonui/NT5design.c:139]:
```suggestion
#if 0
```

File: base/system/logonui/NT5design.c [hunk 2]
```diff
@@ -136,8 +136,9 @@ NT5_DrawBaseBackground(HDC hdcDesktop)
...
     BOOL bRet = FALSE;
-
+    #endif
```
Comment (binarymaster) [inline base/system/logonui/NT5design.c:141]:
```suggestion
#endif
```

File: base/system/logonui/logonui.c [hunk 3]
```diff
@@ -36,51 +36,52 @@ DrawLogoffScreen(HDC hdcMem)
...
+         {
+             /* Check the value */
+             Value = GetULONG(String2);
+             *i += 1;
+         }
+     }
+
+     return Value;
+ }
+ #endif
```
Comment (binarymaster) [inline base/system/logonui/logonui.c:84]: For some reason spaces are added on each line here, it should be unmodified.
```suggestion
static ULONG
GetULONG(LPWSTR String)
{
    UINT i, Length;
    ULONG Value;
    LPWSTR StopString;

    i = 0;
    /* Get the string length */
    Length = (UINT)wcslen(String);

    /* Check the string only consists of numbers */
    while ((i < Length) && ((String[i] < L'0') || (String[i] > L'9'))) i++;
    if ((i >= Length) || ((String[i] < L'0') || (String[i] > L'9')))
    {
        return (ULONG)-1;
    }

    /* Convert it */
    Value = wcstoul(&String[i], &StopString, 10);

    return Value;
}

static ULONG
GetULONG2(LPWSTR String1, LPWSTR String2, PINT i)
{
    ULONG Value;

    /* Check the first string value */
    Value = GetULONG(String1);
    if (Value == (ULONG)-1)
    {
        /* Check the second string value isn't a switch */
        if (String2[0] != L'-')
        {
            /* Check the value */
            Value = GetULONG(String2);
            *i += 1;
        }
    }

    return Value;
}
#endif
```

**Review by binarymaster — COMMENTED** [review]
Macros should start at the beginning of the line, without indentation.

File: base/system/logonui/NT5design.c [hunk 4]
```diff
@@ -136,8 +136,9 @@ NT5_DrawBaseBackground(HDC hdcDesktop)
 {
     HBITMAP hBitmap = NULL;
     HDC hdcMem = NULL;
+#if 0
     BOOL bRet = FALSE;
-
+#endif
```
Comment (HBelusca) [inline base/system/logonui/NT5design.c:141]: This bRet can be fully removed.

File: base/system/logonui/logonui.c [hunk 5]
```diff
@@ -36,7 +36,7 @@ DrawLogoffScreen(HDC hdcMem)
     /* Draw the logoff icon */
     NT5_CreateLogoffScreen(L"Saving your settings...", hdcMem);
 }
-
+#if 0
```
Comment (HBelusca) [inline base/system/logonui/logonui.c:39]:
```suggestion

#if 0
```

File: base/system/logonui/CMakeLists.txt [hunk 6]
```diff
@@ -0,0 +1,5 @@
+
+add_executable(logonui logonui.c logonui.rc NT5design.c)
```
Comment (HBelusca) [inline base/system/logonui/CMakeLists.txt:2]: C files first, then the resources.
```suggestion
add_executable(logonui logonui.c NT5design.c logonui.rc)
```

File: base/system/logonui/CMakeLists.txt [hunk 7]
```diff
@@ -0,0 +1,5 @@
...
+set_module_type(logonui win32gui UNICODE)
+add_importlibs(logonui user32 shell32 gdi32 msimg32 msvcrt kernel32)
```
Comment (HBelusca) [inline base/system/logonui/CMakeLists.txt:4]:
```suggestion
add_importlibs(logonui gdi32 user32 msimg32 shell32 msvcrt kernel32)
```

File: base/system/logonui/NT5design.c [hunk 8]
```diff
@@ -30,7 +30,7 @@ NT5_DrawLogoffCaptionText(LPWSTR lpText,
     LogFont.lfCharSet = DEFAULT_CHARSET;
     LogFont.lfHeight = 22;
     LogFont.lfWeight = 109; // From WinXP disassembly
```
Comment (HBelusca) [inline base/system/logonui/NT5design.c:32]: That comment is cringe.

Comment (SigmaTel71) [reply base/system/logonui/NT5design.c:32]: Well, it was [already like that](https://github.com/reactos/reactos/blob/master/base/system/logonui/NT5design.c#L32) when I found the file. I can remove the comment, if required.

Comment (HBelusca) [reply base/system/logonui/NT5design.c]: Hehe no it was just me reacting to what I saw.

File: base/system/logonui/logonui.c [hunk 9]
```diff
@@ -37,6 +37,8 @@ DrawLogoffScreen(HDC hdcMem)
     NT5_CreateLogoffScreen(L"Saving your settings...", hdcMem);
 }
 
+
```
Comment (HBelusca) [inline base/system/logonui/logonui.c:40]:
```suggestion
```

**Review by HBelusca — APPROVED** [review]

File: base/system/logonui/NT5design.c [hunk 10]
```diff
@@ -30,7 +30,7 @@ NT5_DrawLogoffCaptionText(LPWSTR lpText,
...
-    wcscpy_s(LogFont.lfFaceName, LF_FACESIZE, L"Arial");
+    wcscpy(LogFont.lfFaceName, L"Arial");
```
Comment (HBelusca) [inline base/system/logonui/NT5design.c:33]: Actually I just realized you could use
`StringCchCopyW(LogFont.lfFaceName, _countof(LogFont.lfFaceName), L"Arial");`
(and including the necessary headers)

**Comment by xmine64** [comment]
@SigmaTel71 Are you still working on LogonUI?

**Pull request merged**